### PR TITLE
Event facets

### PR DESCRIPTION
**Pull request by kasperg**
#### Link to issue

https://reload.atlassian.net/browse/DDFFORM-65

#### Description

This PR adds support for filtering events based on branches. This introduces the use of Facets to support filtering.

Please go through the individual commits for additional commentary.

#### Screenshot of the result

Monosnap Events | DPL CMS 2024-03-18 14-04-20

#### Additional comments or questions

This PR depends on https://github.com/danskernesdigitalebibliotek/dpl-design-system/pull/559.

### CODE REVIEW

**Review by kasperbirch1 — APPROVED** [review]
Well done! That you got it solved! I have tested it and it works, but there is a minor CSS error in terms of spacing between dropdowns on my local setup / Brave browser.

image

File: web/themes/custom/novel/templates/views/views-view--events.html.twig [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
-  {{ drupal_block("facet_block:branch") }}
+  <div class="content-list-page__filter">
+    {{ drupal_block("facet_block:event_categories") }}
+  </div>
+  <div class="content-list-page__filter">
+    {{ drupal_block("facet_block:branch") }}
+  </div>
+  <div class="content-list-page__filter">
+    {{ drupal_block("facet_block:date") }}
+  </div>
```
Comment (LasseStaus) [inline web/themes/custom/novel/templates/views/views-view--events.html.twig:12]: Is this ment to be here? I see no mention of date filtering and as far as i can see this is not a facet block on `/admin/config/search/facets`

It's causing a lot of warnings on `/events` when logged in.

**Review by LasseStaus — CHANGES_REQUESTED** [review]
See comment about the date facet block. 🚀

File: web/themes/custom/novel/templates/views/views-view--events.html.twig (hunk 1 above)
Comment (kasperg) [reply web/themes/custom/novel/templates/views/views-view--events.html.twig]: You are quite right. I messed up the juggling between the date facet and the category and branch.

**Review by LasseStaus — APPROVED** [review]
Seems good to me! 🎈

**Pull request merged**